### PR TITLE
Add the /api/v1/status endpoint

### PR DESCRIPTION
**Pull request by odannyc**
To get the system status all you need to do now is:
`/api/v1/status`

And it'll return:

```json
{
  "data": {
    "system_status": "success",
    "system_message": "System operational",
    "favicon": "favicon"
  }
}
```

### CODE REVIEW

File: app/Http/Controllers/Api/GeneralController.php [hunk 1]
```diff
@@ -44,4 +46,48 @@ public function version()
...
+        } elseif (Component::enabled()->notStatus(1)->count() === 0) {
+            // If all our components are ok, do we have any non-fixed incidents?
+            $incidents = Incident::notScheduled()->orderBy('created_at', 'desc')->get();
+            $incidentCount = $incidents->count();
+
+            if ($incidentCount === 0 || ($incidentCount >= 1 && (int) $incidents->first()->status === 4)) {
+                $withData = [
+                    'system_status'  => 'success',
+                    'system_message' => trans_choice('cachet.service.good', $totalComponents),
+                    'favicon'        => 'favicon',
```
Comment (GrahamCampbell) [inline app/Http/Controllers/Api/GeneralController.php]: i'm a bit confused why we need that?

File: app/Http/Controllers/Api/GeneralController.php [hunk 2]
```diff
@@ -44,4 +46,48 @@ public function version()
...
+
+    /**
+     * Show the general system status.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function status()
+    {
+        $totalComponents = Component::enabled()->count();
+        $majorOutages = Component::enabled()->status(4)->count();
```
Comment (GrahamCampbell) [inline app/Http/Controllers/Api/GeneralController.php]: Is this code taken from the other controller? If so, we should probably reuse it, not copy it.

File: app/Http/Controllers/Api/GeneralController.php (hunk 1 above)
Comment (odannyc) [inline app/Http/Controllers/Api/GeneralController.php]: It's the same check that the main status page uses:
`app/composers/StatusPageComposer.php`

Comment (GrahamCampbell) [inline app/Http/Controllers/Api/GeneralController.php]: Yeh, but from the API it doesn't have an actual use case?

Comment (odannyc) [inline app/Http/Controllers/Api/GeneralController.php]: Yeah, on the page: https://social.ringplus.net/categories/community there's a section that checks a cachet status page and just displays the same exact message the cachet status page displays, on the backend of that site it gets all components and loops through them to see if any of them are not functional which then displays the message from there. It'll be better if I can just use this API endpoint to get the message directly. Also check the Issue #1936

**Comment by jbrooksuk** [comment]
The favicon can probably be dropped?

**Comment by GrahamCampbell** [comment]
> The favicon can probably be dropped?
 
See the earlier comments. ;)

**Comment by GrahamCampbell** [comment]
Replaced by https://github.com/CachetHQ/Cachet/pull/1945 I think?

**Pull request closed without merging**